src/Factory/RedirectFactory.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Setono\SyliusRedirectPlugin\Factory;
+
+use Setono\SyliusRedirectPlugin\Model\RedirectInterface;
+use Sylius\Component\Core\Model\ProductInterface;
+use Sylius\Component\Resource\Factory\FactoryInterface;
+use Symfony\Component\Routing\RouterInterface;
+
+final class RedirectFactory implements RedirectFactoryInterface
+{
+    /** @var FactoryInterface */
+    private $decoratedFactory;
+
+    /** @var RouterInterface */
+    private $router;
+
+    public function __construct(FactoryInterface $decoratedFactory, RouterInterface $router)
+    {
+        $this->decoratedFactory = $decoratedFactory;
+        $this->router = $router;
+    }
+
+    public function createNew(): RedirectInterface
+    {
+        /** @var RedirectInterface $redirect */
+        $redirect = $this->decoratedFactory->createNew();
+
+        return $redirect;
+    }
+
+    public function createNewWithValues(
+        string $source,
+        string $destination,
+        bool $permanent = true,
+        bool $only404 = true
+    ): RedirectInterface {
+        $redirect = $this->createNew();
+
+        $redirect->setSource($source);
+        $redirect->setDestination($destination);
+        $redirect->setOnly404($only404);
+        $redirect->setPermanent($permanent);
+        $redirect->enable();
+
+        return $redirect;
+    }
+
+    public function createNewForProduct(
```

Comment:
```suggestion
    public function createNewFromProduct(
```

Usually I would call such a method like this, wdyt?